src/taskmanager/BaseHttpHandler.java

```diff
@@ -0,0 +1,70 @@
+package taskmanager;
+
+import com.sun.net.httpserver.HttpExchange;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class BaseHttpHandler {
+
+    protected void sendText(HttpExchange h, String text) throws IOException {
+        byte[] resp = text.getBytes(StandardCharsets.UTF_8);
+        h.getResponseHeaders().add("Content-Type", "application/json;charset=utf-8");
+        h.sendResponseHeaders(200, resp.length);
+        h.getResponseBody().write(resp);
+        h.close();
+    }
+
+    public void sendNotFound(HttpExchange h) throws IOException {
+        String text = "Задача не найдена.";
+        byte[] resp = text.getBytes(StandardCharsets.UTF_8);
+        h.getResponseHeaders().add("Content-Type", "text/plain;charset=utf-8");
+        h.sendResponseHeaders(404, resp.length);
+        h.getResponseBody().write(resp);
+        h.close();
+    }
+
+    public void sendHasOverlaps(HttpExchange h) throws IOException {
+        String error = "Задача пересекается с другой задачей";
+        byte[] errorBytes = error.getBytes(StandardCharsets.UTF_8);
+        h.getResponseHeaders().add("Content-Type", "text/plain;charset=utf-8");
+        h.sendResponseHeaders(406, errorBytes.length);
+        try (OutputStream os = h.getResponseBody()) {
+            os.write(errorBytes);
+        }
```

Comment:
в методах с try можно закрыть h тоже через finally

```suggestion
        } finally {
            h.close();
        }
```

Reply:
Исправил